src/main/kotlin/com/terraformation/backend/search/field/UriField.kt

```diff
@@ -0,0 +1,51 @@
+package com.terraformation.backend.search.field
+
+import com.terraformation.backend.db.collation
+import com.terraformation.backend.i18n.currentLocale
+import com.terraformation.backend.search.FieldNode
+import com.terraformation.backend.search.SearchFilterType
+import com.terraformation.backend.search.SearchTable
+import java.net.URI
+import java.util.EnumSet
+import org.jooq.Condition
+import org.jooq.Field
+import org.jooq.impl.DSL
+
+/** Search field for urls. */
+class UriField(
+    override val fieldName: String,
+    override val databaseField: Field<URI?>,
+    override val table: SearchTable,
+    override val nullable: Boolean = true,
+) : SingleColumnSearchField<URI>() {
+  override val localize: Boolean
+    get() = false
+
+  override val supportedFilterTypes: Set<SearchFilterType>
+    get() = EnumSet.of(SearchFilterType.Exact, SearchFilterType.PhraseMatch)
+
+  override fun getCondition(fieldNode: FieldNode): Condition {
+    val valuesWithoutScheme =
+        fieldNode.values.map { it?.removePrefix("https://")?.removePrefix("http://") }
+    val uris = valuesWithoutScheme.mapNotNull { URI("https://$it") }
```

Comment:
If someone includes a scheme, why would we want to ignore what they asked us to search for and force it to HTTPS? And if someone doesn't include a scheme, it should be fine because the search is already doing a substring match rather than a prefix match or an exact match.

But it's possible I'm just not seeing the use case here.

Reply:
I was thinking about the case of creating a URI object without a scheme, since the contains operation expects one. 

What I think would suit the use cases is to treat the table field as string. But jooq is giving me a hard time with that. I was looking into some casting or coercing functions but difficult. 

I updated this part to use regex to perform substring matching and this works well so far.

Let me know if you have any advice on treating the table field as a string, which will enable Fuzzy searches as well!